File: Jim-and-the-Orders.cs
using System.CodeDom.Compiler;
using System.Collections.Generic;
using System.Collections;
using System.ComponentModel;
using System.Diagnostics.CodeAnalysis;
using System.Globalization;
using System.IO;
using System.Linq;
using System.Reflection;
using System.Runtime.Serialization;
using System.Text.RegularExpressions;
using System.Text;
using System;

class Result
{

    /*
     * Complete the 'jimOrders' function below.
     *
     * The function is expected to return an INTEGER_ARRAY.
     * The function accepts 2D_INTEGER_ARRAY orders as parameter.
     */

    public static List<int> jimOrders(List<List<int>> orders)
    {
        List<(int serve, int id)> list = new List<(int,int)>();

        for(int i = 0; i < orders.Count; i++)
            list.Add((orders[i][0] + orders[i][1], i + 1));

        list.Sort((a,b) => {
            if(a.serve == b.serve) return a.id.CompareTo(b.id);
            return a.serve.CompareTo(b.serve);
        });

        List<int> result = new List<int>();
        foreach(var x in list)
            result.Add(x.id);

        return result;
    }
    
    
    public static List<int> jimOrders4(List<List<int>> orders)
    {
        var pq = new PriorityQueue<int, (int totalTime, int index)>();

        for (int i = 0; i < orders.Count; i++)
        {
            int totalTime = orders[i][0] + orders[i][1];
            pq.Enqueue(i + 1, (totalTime, i + 1));
        }

        var result = new List<int>();

        while (pq.Count > 0)
        {
            result.Add(pq.Dequeue());
        }

        return result;
    }    
}


class Solution
{
    public static void Main(string[] args)
    {
        TextWriter textWriter = new StreamWriter(@System.Environment.GetEnvironmentVariable("OUTPUT_PATH"), true);

        int n = Convert.ToInt32(Console.ReadLine().Trim());

        List<List<int>> orders = new List<List<int>>();

        for (int i = 0; i < n; i++)
        {
            orders.Add(Console.ReadLine().TrimEnd().Split(' ').ToList().Select(ordersTemp => Convert.ToInt32(ordersTemp)).ToList());
        }

        List<int> result = Result.jimOrders(orders);

        textWriter.WriteLine(String.Join(" ", result));

        textWriter.Flush();
        textWriter.Close();
    }
}

File: Jim-and-the-Orders.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'jimOrders' function below.
     *
     * The function is expected to return an INTEGER_ARRAY.
     * The function accepts 2D_INTEGER_ARRAY orders as parameter.
     */

    public static List<Integer> jimOrders(List<List<Integer>> orders) {
        List<int[]> list = new ArrayList<>();

        for(int i = 0; i < orders.size(); i++)
            list.add(new int[]{orders.get(i).get(0) + orders.get(i).get(1), i + 1});

        Collections.sort(list, (a,b) -> {
            if(a[0] == b[0]) return a[1] - b[1];
            return a[0] - b[0];
        });

        List<Integer> result = new ArrayList<>();
        for(int[] x : list)
            result.add(x[1]);

        return result;
    }
    
    
    public static List<Integer> jimOrders4(List<List<Integer>> orders) {
        PriorityQueue<int[]> pq = new PriorityQueue<>(
            (a, b) -> {
                if (a[0] != b[0]) return a[0] - b[0];
                return a[1] - b[1];
            }
        );

        for (int i = 0; i < orders.size(); i++) {
            int totalTime = orders.get(i).get(0) + orders.get(i).get(1);
            pq.offer(new int[]{totalTime, i + 1});
        }

        List<Integer> result = new ArrayList<>();

        while (!pq.isEmpty()) {
            result.add(pq.poll()[1]);
        }

        return result;
    }    
}


public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));
        BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(System.getenv("OUTPUT_PATH")));

        int n = Integer.parseInt(bufferedReader.readLine().trim());

        List<List<Integer>> orders = new ArrayList<>();

        IntStream.range(0, n).forEach(i -> {
            try {
                orders.add(
                    Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                        .map(Integer::parseInt)
                        .collect(toList())
                );
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        });

        List<Integer> result = Result.jimOrders(orders);

        bufferedWriter.write(
            result.stream()
                .map(Object::toString)
                .collect(joining(" "))
            + "\n"
        );

        bufferedReader.close();
        bufferedWriter.close();
    }
}
